opm/simulators/wells/WellInterfaceFluidSystem.cpp

```diff
@@ -916,6 +918,181 @@ flowPhaseToEbosPhaseIdx(const int phaseIdx) const
     return phaseIdx;
 }
 
+template<typename FluidSystem>
+double
+WellInterfaceFluidSystem<FluidSystem>::
+getGroupInjectionTargetRate(const Group& group,
+                            const WellState& well_state,
+                            const GroupState& group_state,
+                            const Schedule& schedule,
+                            const SummaryState& summaryState,
+                            const InjectorType& injectorType,
+                            double efficiencyFactor,
+                            DeferredLogger& deferred_logger) const
+{
+    // Setting some defaults to silence warnings below.
+    // Will be overwritten in the switch statement.
+    Phase injectionPhase = Phase::WATER;
+    switch (injectorType) {
+    case InjectorType::WATER:
+    {
+        injectionPhase = Phase::WATER;
+        break;
+    }
+    case InjectorType::OIL:
+    {
+        injectionPhase = Phase::OIL;
+        break;
+    }
+    case InjectorType::GAS:
+    {
+        injectionPhase = Phase::GAS;
+        break;
+    }
+    default:
+        // Should not be here.
+        assert(false);
+    }
+
+    auto currentGroupControl = group_state.injection_control(group.name(), injectionPhase);
+    if (currentGroupControl == Group::InjectionCMode::FLD ||
+        currentGroupControl == Group::InjectionCMode::NONE) {
+        if (!group.injectionGroupControlAvailable(injectionPhase)) {
+            return 1.0;
```

Comment:
This, and also line 974, will give a target of 1 cubic meter per second, would it not be better to return an optional double, and set the target only if relevant? It looks like this may be taken from the production variant, which returns a scaling rather than a target (and a 1.0 scaling would be the same as doing nothing).

Reply:
Good catch. It was a bit back-and-forth if the injector rates should be scaled or just set like we do now. Or will update as you suggest.